### PR TITLE
Add ChartClient (v2)

### PR DESCRIPTION
**Pull request by kristw**
🏆 Enhancements

Create `ChartClient` to facilitate calls to various APIs necessary for embeddable charts. 

Refer to this [design doc](https://docs.google.com/document/d/1nKnBxKTOrHW1bqidOE-9cMPH7NZFAYDqLh8DAcVcGVA/edit?usp=sharing)

Working on unit tests.

This PR replaces #44.

### CODE REVIEW

File: packages/superset-ui-chart/src/clients/ChartClient.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        } as RequestConfig)
+        .then(response => response.json as Json);
+
+      /*
+       * If formData is also specified, override API result
+       * with user-specified formData
+       */
+      return input.formData
+        ? promise.then(
+            (dbFormData: object) =>
```
Comment (xtinec) [inline packages/superset-ui-chart/src/clients/ChartClient.ts]: Think we need to destruct the `json` field here to get the form data json.

File: packages/superset-ui-chart/src/clients/ChartClient.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+              ({
+                ...dbFormData,
+                ...input.formData,
+              } as FormData),
+          )
+        : promise.then((dbFormData: object) => dbFormData as FormData);
```
Comment (xtinec) [inline packages/superset-ui-chart/src/clients/ChartClient.ts]: ditto

File: packages/superset-ui-chart/src/clients/ChartClient.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      ? Promise.resolve(input.formData)
+      : Promise.reject(new Error('At least one of sliceId or formData must be specified'));
+  }
+
+  loadQueryData(formData: FormData, options?: RequestConfig): Promise<object> {
+    const buildQuery = getChartBuildQueryRegistry().get(formData.viz_type);
+    if (buildQuery) {
+      return this.client
+        .post({
+          endpoint: '/api/v1/query',
```
Comment (xtinec) [inline packages/superset-ui-chart/src/clients/ChartClient.ts]: Need a `/` behind `query` to hit the server endpoint

File: packages/superset-ui-chart/src/clients/ChartClient.ts (hunk 1 above)
Comment (kristw) [reply packages/superset-ui-chart/src/clients/ChartClient.ts]: never mind. I check the response of `slice_json` and found what you mean

File: packages/superset-ui-chart/src/query/FormData.ts [hunk 4]
```diff
@@ -9,8 +9,17 @@ import { FormDataMetric, MetricKey } from './Metric';
...
 type Metrics = Partial<Record<MetricKey, FormDataMetric | FormDataMetric[]>>;
 
+export type AnnotationLayerMetadata = {
+  name: string;
+  sourceType?: string;
+};
+
 type BaseFormData = {
   datasource: string;
+  // eslint-disable-next-line camelcase
```
Comment (xtinec) [inline packages/superset-ui-chart/src/query/FormData.ts]: [nit] We could disable the camelcase rule for the entire `BaseFormData` block.

**Pull request merged**